### ISSUE TITLE
Jackson annotation @JsonProperty(required=true) does not translate to @Schema(required=true)

### ISSUE THREAD

**Issue by lenalebt** [smallrye/smallrye-open-api#714]
My current classes are annotated with `@JsonProperty(required=true)` Jackson annotations. However, to mark those properties as required in the generated OpenAPI yaml, I need to add a `@Schema(required=true)' annotation. It would be nice if that property could be auto-generated.

Just for more context: I'm using smallrye-openapi together with kotlin, which has specific language features to distinguish between nullable and non-nullable types. Together with a dependency `com.fasterxml.jackson.module:jackson-module-kotlin`, the `@JsonProperty(required=true)` annotations are autogenerated already. Having the `@Schema` annotations also autogenerated would allow you to handle kotlin nullability/non-nullability at least for Jackson users.

**Comment by phillip-kruger** [comment]
@MikeEdgar - w.d.y.t ? We are supporting `@JsonProperty("newName")` so this does make sense

**Comment by MikeEdgar** [comment]
@phillip-kruger - Yes, I agree this makes sense. It could be handled similarly to how bean validation annotations work perhaps. @lenalebt - are your `@JsonProperty` annotations used on constructor parameters? I am not very knowledgeable on Kotlin, so it will be interesting to see how the annotations translate into Java class information. If you can post a sample that demonstrates the source you would like to have scanned, that would help. Thanks

**Comment by lenalebt** [comment]
I'll check where `com.fasterxml.jackson.module:jackson-module-kotlin` actually puts it and will provide a better answer within 2 days (probably won't find time today). Until then: If I provide the annotations myself, I can put it whereever I like. Example:

```
data class Rainbow(
  @field:JsonProperty("big_unicorn") val bigUnicorn: String, //annotates the field
  @JsonProperty("small_unicorn) val smallUnicorn: String, //annotates constructor parameter
  @get:JsonProperty("medium_unicorn") val mediumUnicorn: String, //annotates getter
)
```

However, I'd like to avoid annotating myself as much as I can. For example

```
data class Foo(
  val bar: String?,  //will be annotated `@Nullable` by the kotlin compiler (because of the question mark)
  val baz: String  //with com.fasterxml.jackson.module:jackson-module-kotlin also as `@JsonProperty(required=true)` (because no question mark) and additionally `@NotNull` by the kotlin compiler
)
```

So - I think for the raw kotlin case, independent of jackson, it would already be useful to check those `@Nullable/@NotNull` annotations - and I guess even Java people would like having those reflected in the generated schema, as they also are common - usage depends on the project of course.

More details later on, I'll then also provide package names and exact locations or where annotations are generated.

**Comment by lenalebt** [comment]
Short version: Summary "Jackson annotation @JsonProperty(required=true) does not translate to @Schema(required=true)" still stands, but "Together with a dependency com.fasterxml.jackson.module:jackson-module-kotlin, the @JsonProperty(required=true) annotations are autogenerated already" does not. Jackson is instrumented at runtime. But there are `org.jetbrains.annotations.NotNull` and `org.jetbrains.annotations.Nullable` annotations present from the kotlin compiler.

Imho this splits in 2 issues: One for kotlin support and one for the jackson stuff. I will create another ticket for kotlin support.

TL;DR

Okay, I had a look what actually gets compiled. For this example:

```
data class CustomerPost(val slug: SlugType?, @field:JsonProperty(required = true) val name: String) {}
```
I get this compiled code (had a look with intellij "show bytecode"):
```
public final class com/example/http/CustomerPost {
  // access flags 0x12
  private final Ljava/lang/String; slug
  @Lorg/jetbrains/annotations/Nullable;() // invisible

  // access flags 0x11
  public final getSlug()Ljava/lang/String;
  @Lorg/jetbrains/annotations/Nullable;() // invisible
   L0
    LINENUMBER 23 L0
    ALOAD 0
    GETFIELD com/example/http/CustomerPost.slug : Ljava/lang/String;
    ARETURN
   L1
    LOCALVARIABLE this Lcom/example/http/CustomerPost; L0 L1 0
    MAXSTACK = 1
    MAXLOCALS = 1

  // access flags 0x12
  private final Ljava/lang/String; name
  @Lcom/fasterxml/jackson/annotation/JsonProperty;(required=true)
  @Lorg/jetbrains/annotations/NotNull;() // invisible

  // access flags 0x11
  public final getName()Ljava/lang/String;
  @Lorg/jetbrains/annotations/NotNull;() // invisible
   L0
    LINENUMBER 23 L0
    ALOAD 0
    GETFIELD com/example/http/CustomerPost.name : Ljava/lang/String;
    ARETURN
   L1
    LOCALVARIABLE this Lcom/example/http/CustomerPost; L0 L1 0
    MAXSTACK = 1
    MAXLOCALS = 1

  // compiled from: CustomerResource.kt
}
```
I left out generated toString etc.

Apparently the kotlin module for jackson changes stuff at runtime in addition. This is what I found in their sources:
```
private fun AnnotatedField.hasRequiredMarker(): Boolean? {
        val byAnnotation = (member as Field).isRequiredByAnnotation()
        val byNullability =  (member as Field).kotlinProperty?.returnType?.isRequired()

        return requiredAnnotationOrNullability(byAnnotation, byNullability)
    }

    private fun AccessibleObject.isRequiredByAnnotation(): Boolean? = annotations
        ?.firstOrNull { it.annotationClass == JsonProperty::class }
        ?.let { it as JsonProperty }
        ?.required
   private fun KType.isRequired(): Boolean = !isMarkedNullable
```
This code in the end seems to end up in `com.fasterxml.jackson.databind.introspect.POJOPropertyBuilder` at runtime. So it does *not* add the annotation at compile-time in any way, but instructs jackson to act as if the annotation was present.
